code/modules/network/network.dm

```diff
@@ -0,0 +1,119 @@
+/*
+	Upgraded devices, such as borgs, the AI, or "Hacktools" have several notable features:
+	they are able to see stealthed networks.
+	they are able to pass "-b" as an argument in place of an encryption key to make an attempt at brute-forcing the network. This will take some time, and may trigger the networks security system, but will eventually get you in.
+*/
+
+var/global/list/networks_by_id = list()
+var/global/list/datum/network/networks_by_wide = list()
+var/global/list/active_network_ids = list()
+
+proc/clean_network_command(command = "")
+	var/list/butchered = explode_text(command, " -") // allows for arguments like {text1 = "hello "} {text2 = "world"} to be caused when executing something like {print -text1 = "hello " -text2 = "world"}
+	command = butchered[1]
+	var/list/arguments = list()
+	for(var/i=2, i<=butchered.len, i++)
+		arguments[i-1] = butchered[i]
+	return list(command, arguments)
+
+/datum/network_argument
+	var/main_command
+	var/list/args = list()
```

Comment:
please rename this to ```net_args``` I am increasingly worried a Human/DM will confuse this with the inbuild proc var ```args```